apps/io-services-cms-webapp/src/webservice/controllers/__test__/regenerate-service-keys.test.ts

```diff
@@ -0,0 +1,277 @@
+import { ApiManagementClient } from "@azure/arm-apimanagement";
+import { Container } from "@azure/cosmos";
+import {
+  ServiceLifecycle,
+  ServicePublication,
+  stores,
+} from "@io-services-cms/models";
+import {
+  RetrievedSubscriptionCIDRs,
+  SubscriptionCIDRsModel,
+} from "@pagopa/io-functions-commons/dist/src/models/subscription_cidrs";
+import { UserGroup } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/azure_api_auth";
+import { NonNegativeInteger } from "@pagopa/ts-commons/lib/numbers";
+import {
+  IPatternStringTag,
+  NonEmptyString,
+} from "@pagopa/ts-commons/lib/strings";
+import * as TE from "fp-ts/lib/TaskEither";
+import request from "supertest";
+import { afterEach, describe, expect, it, vi } from "vitest";
+import { IConfig } from "../../../config";
+import {
+  getSubscription,
+  regenerateSubscriptionKey,
+} from "../../../lib/clients/apim-client";
+import { createWebServer } from "../../index";
+
+vi.mock("../../../lib/clients/apim-client", async () => {
+  // vitest partial mocking
+  const apimClient = (await vi.importActual(
+    "../../../lib/clients/apim-client"
+  )) as unknown as any;
+
+  return {
+    ...apimClient,
+    getSubscription: vi.fn((_) => TE.right(anApimResource)),
+    regenerateSubscriptionKey: vi.fn((_) => TE.right(anApimResource)),
+  };
+});
+
+const apiBasePath = "/api/services/aServiceId/keys/";
+const apiFullPath = `${apiBasePath}primary`;
+const aManageSubscriptionId = "MANAGE-123";
+const ownerId = "123";
+const primaryKey = "any-primary-key-value";
+const secondaryKey = "any-secondary-key-value";
+
+const anApimResource = {
+  id: "any-id",
+  name: "any-name",
+  _etag: "_etag",
+  ownerId,
```

Comment:
When apim is called the returned object as the property `ownerId` valued in this format "something/.../userId", is this handled correctly?